### PR TITLE
Django upgrade + design system implementation

### LINKED ISSUE #1349: Upgrade python/django/etc versions
It would be good to upgrade (post elections)
Python and Django versions - there are various libs that are blocked on this and it will probably be a complimentary piece of work to #1283. 
I'm creating this issue to keep track of other libraries that are blocked by this.

### PR DESCRIPTION
**Pull request by VirginiaDooley**
Closes #1349 and #1283

- Upgrade to Django 4.1 with related dependency upgrades
- Implement the design system with related dependency upgrades. 
- Address related deprecation warnings and already deprecated features
- Refactor id_creator following removal of dc_base_theme. Ref https://github.com/DemocracyClub/EveryElection/issues/1576
- Update CircleCI with current image requirements

### CODE REVIEW

File: every_election/apps/organisations/migrations/0063_alter_divisiongeography_geography.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 3.2 on 2022-08-11 15:45
```
Comment (GeoWill) [inline every_election/apps/organisations/migrations/0063_alter_divisiongeography_geography.py:1]: I'm a bit confused about what this migration is doing. It seems to duplicate what is done in every_election/apps/organisations/migrations/0024_auto_20170210_1537.py.

**Comment by GeoWill** [comment]
This is really exciting! 

I had a go at checking it out and ran into a couple issues. 

- I tried installing the fix to dc_base_theme with `pip install git+https://github.com/DemocracyClub/dc_base_theme.git@dbb31123cd983349be0a8168bfa14fa8fd7e9f22` but the test at ` every_election/apps/elections/tests/test_id_creator_wizard.py` still fails. 

- When I `server` to front page looks a bit funny: Screenshot from 2022-09-01 08-15-53

**Comment by symroe** [comment]
> front page looks a bit funny

This is on my advice to ignore the frontend until tests / dependencies are all working. Reason being is that we didn't want to alter the `dc_base_theme` too much, as we're moving to the design system anyway. 

I think we could move on to implementing the design system as part of this PR, or as a different PR after this, but it's useful to just focus on the "backend" work first.

File: requirements/base.txt [hunk 2]
```diff
@@ -32,5 +32,4 @@ importlib-metadata==4.8.1
 
 git+https://github.com/mysociety/django-pipeline-csscompressor
 git+https://github.com/DemocracyClub/dc_base_theme.git@0.3.16
-git+https://github.com/DemocracyClub/dc_signup_form.git@2.1.1
-git+https://github.com/DemocracyClub/uk-geo-utils.git@0.10.5
```
Comment (symroe) [inline requirements/base.txt]: Is this removing geo_utils?

Comment (VirginiaDooley) [reply requirements/base.txt]: No, I thought it was duplicated on line 27?

File: every_election/apps/organisations/migrations/0063_alter_divisiongeography_geography.py (hunk 1 above)
Comment (VirginiaDooley) [reply every_election/apps/organisations/migrations/0063_alter_divisiongeography_geography.py:1]: Could it be addressing a change in `elections`?

`makemigrations` output: 

`Migrations for 'elections':
  every_election/apps/elections/migrations/0063_alter_explanation_explanation.py
    - Alter field explanation on explanation
Migrations for 'organisations':
  every_election/apps/organisations/migrations/0063_alter_divisiongeography_geography.py
    - Alter field geography on divisiongeography`

**Comment by VirginiaDooley** [comment]
> f213ef



> This is really exciting!
> 
> I had a go at checking it out and ran into a couple issues.
> 
> * I tried installing the fix to dc_base_theme with `pip install git+https://github.com/DemocracyClub/dc_base_theme.git@dbb31123cd983349be0a8168bfa14fa8fd7e9f22` but the test at ` every_election/apps/elections/tests/test_id_creator_wizard.py` still fails.
> * When I `runserver` to front page looks a bit funny: Screenshot from 2022-09-01 08-15-53

`design system` now implemented and failing test addressed with https://github.com/DemocracyClub/EveryElection/pull/1556/commits/7f213ef57a6c2d7c7397f8d21c41dca8ecf24860

**Comment by symroe** [comment]
Just looking a little more into this and I found that there are some remains of the foundation HTML left over. Search in HTML files for `columns` and you'll see some things like 
```html 
<div class="row">
<section class="columns large-10 large-centered">
```

I think this is only in a couple of templates, but could do with being removed :+1:

File: every_election/templates/base.html [hunk 3]
```diff
@@ -5,150 +5,116 @@
...
+      </div>
     </div>
-  </div>
-</footer>
-{% endblock site_footer %}
-
+  </footer>
+{% endblock footer_base %}
 
 {% block extra_javascript %}
```
Comment (symroe) [inline every_election/templates/base.html:94]: This block has been renamed to `extra_site_js`

File: every_election/apps/elections/templates/id_creator/date.html [hunk 4]
```diff
@@ -8,38 +8,44 @@ <h1>Poll date</h1>
 {% block extra_javascript %}
 {{ block.super }}
```
Comment (symroe) [inline every_election/apps/elections/templates/id_creator/date.html:9]: We now have `in_page_javascript` that might be better to use here, as that would save the call to `super`

**Review by symroe — APPROVED** [review]

**Comment by VirginiaDooley** [comment]
[Bump dc_utils to version 2.1 in all projects](https://trello.com/c/oyrIDWEA/3048-bump-dcutils-to-version-21-in-all-projects)

**Pull request merged**